extensions/wikia/FounderProgressBar/FounderProgressBarHooks.class.php

```diff
@@ -198,24 +198,29 @@ public static function bonusTaskEnabled($task_id) {
 
 	// Initialize a scratch record for each of the initial available tasks
 	public static function initRecords($wiki_id) {
+		global $wgExternalSharedDB, $wgMemc;
 		wfProfileIn(__METHOD__);
 
-		// Records go into global wikicites table
-		$app = F::app();
-		$dbw = wfGetDB(DB_MASTER, array(), $app->wg->ExternalSharedDB);
+		try {
+			// Records go into global wikicities table
+			$dbw = wfGetDB( DB_MASTER, [], $wgExternalSharedDB );
 
-		foreach(FounderProgressBarController::$tasks as $task_id) {
-			if($task_id < FounderProgressBarController::REGULAR_TASK_MAX_ID) {
-				$sql = "INSERT IGNORE INTO founder_progress_bar_tasks SET wiki_id=$wiki_id, task_id=$task_id";
-				$dbw->query ($sql, __METHOD__);
+			foreach ( FounderProgressBarController::$tasks as $task_id ) {
+				if ( $task_id < FounderProgressBarController::REGULAR_TASK_MAX_ID ) {
+					$sql = sprintf( "INSERT INTO founder_progress_bar_tasks SET wiki_id=%d, task_id=%d",
```

Comment:
Can't we simply use `$dbw->insert()`?
It's widely used and protected from sqlinjections.

Reply:
Also to improve performance we can make this operation in one SQL statement.

Reply:
Changed in 3b4d472